aviatrix/data_source_aviatrix_all_transit_gateways.go

```diff
@@ -0,0 +1,557 @@
+		transitGateway["enable_segmentation"] = gw.EnableSegmentation
+		transitGateway["enable_active_standby"] = gw.EnableActiveStandby
+		transitGateway["enable_active_standby_preemptive"] = gw.EnableActiveStandbyPreemptive
+		transitGateway["enable_transit_summarize_cidr_to_tgw"] = gw.EnableTransitSummarizeCidrToTgw
+
+		if gw.EnableTransitFirenet && goaviatrix.IsCloudType(gw.CloudType, goaviatrix.GCPRelatedCloudTypes) {
+			transitGateway["lan_vpc_id"] = gw.BundleVpcInfo.LAN.VpcID
+			transitGateway["lan_private_subnet"] = strings.Split(gw.BundleVpcInfo.LAN.Subnet, "~~")[0]
+		}
+
+		var prependAsPath []string
+		for _, p := range strings.Split(gw.PrependASPath, " ") {
+			if p != "" {
+				prependAsPath = append(prependAsPath, p)
+			}
+		}
+
+		transitGateway["prepend_as_path"] = prependAsPath
+		transitGateway["enable_monitor_gateway_subnets"] = gw.MonitorSubnetsAction == "enable"
+		transitGateway["monitor_exclude_list"] = gw.MonitorExcludeGWList
+
+		transitGateway["enable_bgp_over_lan"] = goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AzureArmRelatedCloudTypes|goaviatrix.GCPRelatedCloudTypes) && gw.EnableBgpOverLan
+		if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.GCPRelatedCloudTypes) && gw.EnableBgpOverLan {
+			if len(gw.BgpLanInterfaces) != 0 {
+				var interfaces []map[string]interface{}
+				for _, bgpLanInterface := range gw.BgpLanInterfaces {
+					interfaceDict := make(map[string]interface{})
+					interfaceDict["vpc_id"] = bgpLanInterface.VpcID
+					interfaceDict["subnet"] = bgpLanInterface.Subnet
+					interfaces = append(interfaces, interfaceDict)
+				}
+				transitGateway["bgp_lan_interfaces"] = interfaces
+			}
+
+			if len(gw.HaGw.HaBgpLanInterfaces) != 0 {
+				var haInterfaces []map[string]interface{}
+				for _, haBgpLanInterface := range gw.HaGw.HaBgpLanInterfaces {
+					interfaceDict := make(map[string]interface{})
+					interfaceDict["vpc_id"] = haBgpLanInterface.VpcID
+					interfaceDict["subnet"] = haBgpLanInterface.Subnet
+					haInterfaces = append(haInterfaces, interfaceDict)
+				}
+				transitGateway["ha_bgp_lan_interfaces"] = haInterfaces
+			}
+		} else {
+			transitGateway["bgp_lan_ip_list"] = nil
+			transitGateway["ha_bgp_lan_ip_list"] = nil
+		}
+
+		if gw.EnableSpotInstance {
+			transitGateway["enable_spot_instance"] = true
+			transitGateway["spot_price"] = gw.SpotPrice
+		}
+
+		if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.AzureArmRelatedCloudTypes) {
+			azureEip := strings.Split(gw.ReuseEip, ":")
+			if len(azureEip) == 3 {
+				transitGateway["azure_eip_name_resource_group"] = fmt.Sprintf("%s:%s", azureEip[0], azureEip[1])
+			} else {
+				log.Printf("[WARN] could not get Azure EIP name and resource group for the Transit Gateway %s", gw.GwName)
+			}
+		}
+
+		result = append(result, transitGateway)
+	}
+
+	if err = d.Set("transit_gateway_list", result); err != nil {
+		return fmt.Errorf("couldn't set transit_gateway_list: %s", err)
+	}
+	d.SetId("transit-gateway-list-id")
```

Comment:
The ID for this data source should probably be the Controller IP. We want the data source to be unique per Controller, but not necessarily unique per configuration. You can refer to `aviatrix_controller_bgp_max_as_limit_config` as an example of how to set ID.